src/main/java/jenkins/bouncycastle/api/PEMEncodable.java

```diff
@@ -140,9 +144,23 @@ public static PEMEncodable decode(@Nonnull String pem, @Nullable final char[] pa
 
         PEMReader parser = new PEMReader(new StringReader(pem), pwf);
         try {
-            return new PEMEncodable(parser.readObject());
-        } catch (PasswordException pwE) {
+            Object object = parser.readObject();
+
+            // JENKINS-35661 in this case we know how to get the public key too
+            if (object instanceof RSAPrivateCrtKey) {
+                // obtain public key spec from the private key
+                RSAPrivateCrtKey rsaPK = (RSAPrivateCrtKey) object;
+                RSAPublicKeySpec pubKeySpec = new RSAPublicKeySpec(rsaPK.getModulus(), rsaPK.getPublicExponent());
+                KeyFactory kf = KeyFactory.getInstance("RSA");
+                return new PEMEncodable(new KeyPair(kf.generatePublic(pubKeySpec), rsaPK));
+            }
+
+            return new PEMEncodable(object);
+        } catch (PasswordException | InvalidKeySpecException pwE) {
```

Comment:
This branch has 1.609 as core dependency. IIRC that core version was supposed to support Java 6. So multi-catch syntax is not allowed.

BTW the `java.level` should be set to 6 in this branch.

Info [here](https://jenkins.io/blog/2015/04/06/good-bye-java6/).